content/developer/reference/frontend/owl_components.rst

```diff
@@ -150,13 +150,138 @@ checkboxes or datepickers. This page explains how to use these generic component
      - full-featured dropdown
    * - :ref:`Pager <frontend/pager>`
      - a small component to handle pagination
 
+.. _frontend/owl/actionswiper:
+
+ActionSwiper
+------------
+
+.. note:: This component is a mobile feature, only supported in the Enterprise version of Odoo.
+
+Location
+~~~~~~~~
+
+`@web_enterprise/core/action_swiper/action_swiper`
+
+Description
+~~~~~~~~~~~
+
+This is a component that can perform actions when an element is swiped
+horizontally. The swiper is wrapping a target element to add actions to it. 
+The action is executed once the user has released the swiper passed
+half of its width.
+
+.. code-block:: xml
+
+  <ActionSwiper onLeftSwipe="Object" onRightSwipe="Object">
+    <SomeElement/>
+  </ActionSwiper>
+
+The simplest way to use the component is to use it around your target element directly
+in an xml template as shown above. But sometimes, you may want to extend an existing element
+and would not want to duplicate the template. It is possible to do just that.
+
+If you want to extend the behavior of an existing element, you must place the element
+inside, by wrapping it directly. Also, you can conditionnally add props to manage when the
+element might be swipable or not. 
+
+You can use the component to interact easily with records, messages, items in lists and much more.
+
+.. image:: ./images/actionswiper.png
+  :width: 400 px
+  :alt: Example of ActionSwiper usage
+  :align: center
+
+The following example creates a basic ActionSwiper component. 
+Here, the swipe is enabled in both directions.
+
+.. code-block:: xml
+    
+  <ActionSwiper 
+    onRightSwipe="
+      {
+        action: '() => Delete item',
+        icon: 'fa-delete',
+        bgColor: 'bg-danger',
+      }" 
+    onLeftSwipe="
+      {
+        action: '() => Star item',
+        icon: 'fa-star',
+        bgColor: 'bg-warning',
+      }"
+  >
+    <div>
+      Swipable item
+    </div>
+  </ActionSwiper>
+
+.. note:: Actions are permuted when using right-to-left (RTL) languages.
+
+Props
+~~~~~
+
+.. list-table::
+    :widths: 20 20 60
+    :header-rows: 1
+
+    * - Name 
+      - Type
+      - Description
+    * - `onLeftSwipe`
+      - `Object`
+      - if present, the actionswiper can be swiped to the left
+    * - `onRightSwipe`
+      - `Object`
+      - if present, the actionswiper can be swiped to the right
+
+You can use both `onLeftSwipe` and `onRightSwipe` props at the same time.
+
+Those `Object`'s must contain:
+    
+    - `action`, which is the callable `Function` serving as a callback.
+      Once the swipe has been completed in the given direction, that action
+      is performed.
+    - `icon` is the icon class to use, usually to represent the action.
+      It must be a `string`.
+    - `bgColor` is the background color, given to decorate the action.
+      can be one of the following `bootstrap contextual color <https://getbootstrap.com/docs/3.3/components/#available-variations>`_ (`danger`,
```

Comment:
I don't usually comment on line length on JS PRs but, just in case, you can totally do this:
```suggestion
      can be one of the following `bootstrap contextual color
      <https://getbootstrap.com/docs/3.3/components/#available-variations>`_ (`danger`,
```